### PR TITLE
Enable widget load assets in its own package in test

### PR DESCRIPTION
**Pull request by chunhtai**
## Description

If a package has a widget that depends on assets in its own package, it will have to load it with the packages/<package_name>/ prefix. It will fail if the test in the package try to test the widget, since all the assets keys are stored as relative path in the package pubspec.yaml.

This pr do a special look up during the test to make sure it cover this case.

## Related Issues

https://github.com/flutter/flutter/issues/32609

## Tests

I added the following tests:

"can load assets within its own package"

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by chunhtai** [comment]
@jonahwilliams Would you mind review this pr? For some reason i couldn't request you as reviewer

File: packages/flutter_test/lib/src/binding.dart [hunk 1]
```diff
@@ -811,15 +811,21 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     final String assetFolderPath = Platform.environment['UNIT_TEST_ASSETS'];
     _ensureInitialized(assetFolderPath);
 
+    final String prefix =  'packages/${Platform.environment['APP_NAME']}/';
```
Comment (jonahwilliams) [inline packages/flutter_test/lib/src/binding.dart]: Add a short comment here explaining the issue this solves

**Review by jonahwilliams — APPROVED** [review]
LGTM

Would like a comment in the flutter_test binding explaining things a bit more

**Comment by chunhtai** [comment]
> LGTM
> 
> Would like a comment in the flutter_test binding explaining things a bit more

I would like to point out a little decision I made.

According to this document https://api.flutter.dev/flutter/widgets/Image/Image.asset.html
we mentioned that "The package argument must be non-null when displaying an image from a package and null otherwise. See the Assets in packages section for details."

This pr not only enables package to loads its packages with package argument, but it also enables flutter app to do the same thing in test as well. Which might result in a false positive test. I personally think this is ok. However, I would like to know if others are ok with this.

There are two other alternative.
1, figure out if this is a package test or app test during unit test assets setup, I am not sure if there is a good way to do this.
2, Let app to be able to load with its own package name specify. This does however will promise a different things then the api document

**Pull request merged**